src/irmin-pack/layout.ml

```diff
@@ -40,8 +40,14 @@ module V3 = struct
   let sorted ~generation =
     toplevel ("store." ^ string_of_int generation ^ ".sorted")
 
-  let mapping ~generation =
-    toplevel ("store." ^ string_of_int generation ^ ".mapping")
+  let mapping =
+    (* if we use mmap for the mapping file, then the mapping file is not portable between
+       architectures with different endianness; to avoid the possibility that a
+       little-endian file is used on a big-endian arch (or vice versa) we use a filename
+       that is not portable between archs *)
+    let endian = match Sys.big_endian with true -> "be" | false -> "le" in
```

Comment:
could we have an empty suffix for the most common case?

Reply:
Great suggestion! I'll change this now...